### PR TITLE
US020 - Delete a role (front-end)

### PR DESCRIPTION
**Pull request by ManpreetM01**

### CODE REVIEW

File: src/service/jobRoleService.ts [hunk 1]
```diff
@@ -12,5 +12,29 @@ export const jobRoleService = {
...
+    async deleteJobRole(id: number): Promise<void>{
+        try {
+            const response = await axios.delete(process.env.API_URL + "api/job-roles/" + id);
+            if (response.status === 200) {
+                return response.data
+            }
+        }
+        catch (e) {
+            throw new Error("Failed to delete job role")
+        }
```
Comment (TomBillingtonUK) [inline src/service/jobRoleService.ts:40]: You may want to log the error messages to the console so you know what the issue is if it doesn't work

File: test/ui/viewJobRolesTest.ts [hunk 2]
```diff
@@ -13,6 +13,74 @@ describe('job-roles page', () => {
...
+
+        const deleteButton = await driver.findElement(webdriver.By.className('btn-primary'));
+        await deleteButton.click();
+
+        await driver.get(process.env.UI_TEST_URL + '/view-job-role/1');
+        const error = await driver.findElement(webdriver.By.css('div.alert.alert-danger')).getText();
+        expect(error).to.equal('Error: Failed to get job role');
+
+        await driver.quit(); 
+    })
```
Comment (TomBillingtonUK) [inline test/ui/viewJobRolesTest.ts]: If you delete job role one in the database on the first run, how does this work if I run the test a second time

File: views/delete-job-role.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{% extends "templates/master.html" %}
```
Comment (TomBillingtonUK) [inline views/delete-job-role.html:1]: Can I have some screenshots of the UI please

Comment (ManpreetM01) [reply views/delete-job-role.html:1]: Screenshot 2023-09-21 at 12 10 03
5/fb7bb610-fffb-4c8c-bce5-5a17663ad53c">
c69-4f34-8693-233fb7e9cd68">

Comment (ManpreetM01) [reply views/delete-job-role.html:1]: Screenshot 2023-09-21 at 12 10 03

Comment (ManpreetM01) [reply views/delete-job-role.html:1]: Screenshot 2023-09-21 at 12 10 40

Comment (ManpreetM01) [reply views/delete-job-role.html:1]: Screenshot 2023-09-21 at 12 10 54

**Review by TomBillingtonUK — APPROVED** [review]

**Review by anthony-raybould-kainos — APPROVED** [review]
Looks good, just remember to change the UI tests to get their driver from the buildDriver function when merging into main, and check that everything still works with the authorisation in main.

**Pull request merged**